### PR TITLE
Restructured Java samples

### PR DESCRIPTION
**Pull request by clemensv**
Restructured the Java samples, breaking the single project apart in different Maven projects and adding an umbrella Maven build. Added an ARM template to deploy a base topology for the services. Added helper scripts. Added readme.md for all samples.

### CODE REVIEW

File: samples/Java/azure-servicebus/QueueClientQuickstart/src/main/java/com/microsoft/azure/servicebus/samples/queueclientquickstart/QueueClientQuickstart.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        // command line to continue
+        if (!parseCommandLine(args)) {
+            return;
+        }
+
+        logger.info("Starting BasicSendReceiveWithQueueClient sample");
+
+        // Create a QueueClient instance using the connection string builder
+        logger.info("Create queue client.");
+        queueClient = new QueueClient(new ConnectionStringBuilder(connectionString, queueName), ReceiveMode.PEEKLOCK);
```
Comment (nemakam) [inline samples/Java/azure-servicebus/QueueClientQuickstart/src/main/java/com/microsoft/azure/servicebus/samples/queueclientquickstart/QueueClientQuickstart.java]: >ReceiveMode.PEEKLOCK [](start = 96, length = 20)

Given that this is the first sample in this tutorial/sample, maybe a small description of what `PeekLock` is might be easier for users. Difference between `PeekLock` and `ReceiveAndDelete` and probably what is `AutoComplete`

File: samples/Java/azure-servicebus/QueueClientQuickstart/src/main/java/com/microsoft/azure/servicebus/samples/queueclientquickstart/QueueClientQuickstart.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+                return CompletableFuture.completedFuture(null);
+            }
+
+            // callback invoked when the message handler has an exception to report
+            public void notifyException(Throwable throwable, ExceptionPhase exceptionPhase) {
+                logger.error(exceptionPhase + "-" + throwable.getMessage());
+            }
+        },
+                // 1 concurrent call, messages are auto-completed, auto-renew duration
+                new MessageHandlerOptions(1, true, Duration.ofMinutes(1)));
```
Comment (nemakam) [inline samples/Java/azure-servicebus/QueueClientQuickstart/src/main/java/com/microsoft/azure/servicebus/samples/queueclientquickstart/QueueClientQuickstart.java]: >true, Duration.ofMinutes(1) [](start = 45, length = 27)

Maybe we should add a comment here for each of the args?

File: samples/Java/azure-servicebus/TopicClientQuickstart/readme.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# Topic Client Quickstart
+
+This sample demonstrates how to use Azure Service Bus Topics with the Azure Service Bus SDK for Java.
+
+You will learn how to set up a TopicClient and send messages, and hwo to set up a SubscriptionClient and
```
Comment (nemakam) [inline samples/Java/azure-servicebus/TopicClientQuickstart/readme.md]: >hwo [](start = 66, length = 3)

typo

File: samples/Java/azure-servicebus/TopicClientQuickstart/readme.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
+# Topic Client Quickstart
```
Comment (nemakam) [inline samples/Java/azure-servicebus/TopicClientQuickstart/readme.md]: Could you add a small comment on difference between a queue and a topic-subscription?

File: samples/Java/azure-servicebus/QueueClientQuickstart/src/main/java/com/microsoft/azure/servicebus/samples/queueclientquickstart/QueueClientQuickstart.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        logger.info("Create queue client.");
+        queueClient = new QueueClient(new ConnectionStringBuilder(connectionString, queueName), ReceiveMode.PEEKLOCK);
+
+        // send messages in a loop
+        for (int i = 0; i < totalToSend; i++) {
+            int currentMessageCounter = i;
+            queueClient.sendAsync(new Message("" + i)).thenRunAsync(() -> {
+                logger.info(String.format("Sent message #%d.", currentMessageCounter));
+            });
+        }
```
Comment (nemakam) [inline samples/Java/azure-servicebus/QueueClientQuickstart/src/main/java/com/microsoft/azure/servicebus/samples/queueclientquickstart/QueueClientQuickstart.java]: Given that the aim of these samples also include best practices, we should have these inside `try-catch` and have `queueClient.close()` in `finally`

File: samples/Java/azure-servicebus/TopicClientQuickstart/src/main/java/com/microsoft/azure/servicebus/samples/topicclientquickstart/TopicClientQuickstart.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+public class TopicClientQuickstart {
+    // Connection String for the namespace can be obtained from the Azure portal under the
+    // 'Shared Access policies' section.
+    private static String connectionString = System.getenv("SB_SAMPLES_CONNECTIONSTRING");
+    private static String topicName = System.getenv("SB_SAMPLES_TOPICNAME");
+    private static String subscriptionName = System.getenv("SB_SAMPLES_SUBSCRIPTIONNAME");
+
+    private static ITopicClient topicClient;
+    private static ISubscriptionClient subscriptionClient;
+    private static int totalSend = 100;
```
Comment (nemakam) [inline samples/Java/azure-servicebus/TopicClientQuickstart/src/main/java/com/microsoft/azure/servicebus/samples/topicclientquickstart/TopicClientQuickstart.java]: >100 [](start = 35, length = 3)

Maybe change to 10? Too many logger messages (send/receive) on the screen might intimidate the first time user.

File: samples/Java/qpid-jms-client/JmsQueueQuickstart/readme.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+This sample demonstrates how to use Azure Service Bus Queues with the Java Message Service (JMS) API, 
+implemented via the AMQP 1.0 to JMS mapping provcided by the Apache Qpid JMS client. The Apache Qpid 
+JMS client is maintained by the Apache Foundation's Qpid Proton project.
+
+Using Azure Service Bus queues through the JMS API provides basic send and receive capabilities and is 
+a convenient choice when porting applications from other message brokers with JMS compliant APIs.
+
+For cloud-native applications and applications that want to take advantage of more advanced Service Bus 
+capabilities, teh native Service Bus API provides deeper and more direct access to the service
```
Comment (nemakam) [inline samples/Java/qpid-jms-client/JmsQueueQuickstart/readme.md]: >teh [](start = 14, length = 3)

nit

File: samples/Java/azure-servicebus/MessageReceiverQuickstart/readme.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
+# Message Receiver Quickstart
+
+This sample demonstrates how to use Azure Service Bus Queues with the Azure Service Bus SDK for Java.
+
+You will learn how to set up a QueueClient, send messages, and receive those messages by explicitly pulling
```
Comment (nemakam) [inline samples/Java/azure-servicebus/MessageReceiverQuickstart/readme.md]: >QueueClient [](start = 31, length = 11)

Sender/Receiver?

File: samples/Java/azure-servicebus/QueueClientQuickstart/src/main/java/com/microsoft/azure/servicebus/samples/queueclientquickstart/QueueClientQuickstart.java (hunk 1 above)
Comment (clemensv) [reply samples/Java/azure-servicebus/QueueClientQuickstart/src/main/java/com/microsoft/azure/servicebus/samples/queueclientquickstart/QueueClientQuickstart.java]: We're currently staging a big update to the conceptual Service Bus docs. I actually want to link to the concepts from the sample comments, probably with shortened URLs. I'll add a comment.

File: samples/Java/azure-servicebus/TopicClientQuickstart/readme.md (hunk 4 above)
Comment (clemensv) [reply samples/Java/azure-servicebus/TopicClientQuickstart/readme.md]: For that I do assume people have read at least the intro portion of the docs and I don't need to repated that here.

File: samples/Java/azure-servicebus/QueueClientQuickstart/src/main/java/com/microsoft/azure/servicebus/samples/queueclientquickstart/QueueClientQuickstart.java (hunk 5 above)
Comment (clemensv) [reply samples/Java/azure-servicebus/QueueClientQuickstart/src/main/java/com/microsoft/azure/servicebus/samples/queueclientquickstart/QueueClientQuickstart.java]: This is the Quickstart. Other samples should be more robust, but it's ok for this one to be a bit more optimistic. It's sending messages inside of a loop and that, per-se, probably isn't "best practice". Adding proper handling here means that you put a try/catch with logging and finally inside that loop and then it gets all a bit hard to read for beginners.

File: samples/Java/qpid-jms-client/JmsQueueQuickstart/readme.md (hunk 7 above)
Comment (clemensv) [reply samples/Java/qpid-jms-client/JmsQueueQuickstart/readme.md]: ok

File: samples/Java/azure-servicebus/TopicClientQuickstart/src/main/java/com/microsoft/azure/servicebus/samples/topicclientquickstart/TopicClientQuickstart.java (hunk 6 above)
Comment (clemensv) [reply samples/Java/azure-servicebus/TopicClientQuickstart/src/main/java/com/microsoft/azure/servicebus/samples/topicclientquickstart/TopicClientQuickstart.java]: ok

File: samples/Java/azure-servicebus/TopicClientQuickstart/readme.md (hunk 3 above)
Comment (clemensv) [reply samples/Java/azure-servicebus/TopicClientQuickstart/readme.md]: ok

File: samples/Java/azure-servicebus/QueueClientQuickstart/src/main/java/com/microsoft/azure/servicebus/samples/queueclientquickstart/QueueClientQuickstart.java (hunk 2 above)
Comment (clemensv) [reply samples/Java/azure-servicebus/QueueClientQuickstart/src/main/java/com/microsoft/azure/servicebus/samples/queueclientquickstart/QueueClientQuickstart.java]: there is a comment for each of the args :)

File: samples/Java/azure-servicebus/MessageReceiverQuickstart/readme.md (hunk 8 above)
Comment (clemensv) [reply samples/Java/azure-servicebus/MessageReceiverQuickstart/readme.md]: ok

**Comment by clemensv** [comment]
Closing and submitting integrated with PR #90

**Pull request closed without merging**